src/lib.rs

```diff
@@ -342,7 +370,11 @@ pub fn parse_beat(text: &str) -> Result<Vec<Cmd>, &str> {
                 }
             }
         })
-        .collect()
+        .collect();
+    match cmds {
```

Comment:
This is `match` could be `.and_then(Code::new)`.

Reply:
… but also if we're going to be able to parse semantically invalid code, then we don't need that anyway.